src/Native/Unix/System.Security.Cryptography.Native/openssl.c

```diff
@@ -736,7 +736,8 @@ static int CheckX509HostnameMatch(ASN1_STRING* candidate, const char* hostname,
         {
             char c = candidateStr[i];
 
-            if ((c < 'a' || c > 'z') && (c < '0' || c > '9') && (c != '.') && (c != '-') && (c != '*' || i != 0))
+            if ((c < 'A' || c > 'Z') && (c < 'a' || c > 'z') && (c < '0' || c > '9') && (c != '.') && (c != '-') &&
+                (c != '*' || i != 0))
```

Comment:
FWIW I always find AND (NOT x OR NOT y) harder to read than AND NOT (x AND y) 

Given `(c != '*' || i != 0)` I need to mentally decode whereas `!(c == '*' && i == 0)` naturally reads as "no asterisk at the front".